### PR TITLE
Implement pagination for getStoryTranslations query

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

[Paginate getStoryTranslations](https://www.notion.so/uwblueprintexecs/Paginate-getStoryTranslations-d86ce7ed0c904d469226b785af549c98)

## Implementation description

- Used graphene relay to add pagination by creating a Connection and modifying the `story_translations` query in `schema.py`
- Created a new object type called `StoryTranslationNode` that represents the returned object for a story translation

## Steps to test

1. Using the Altair client, using the following mutation to log in, then copy the access token and set the authorization header:
```
mutation Login {
  login (email:"planetread+angelamerkel@uwblueprint.org", password:".") {
    accessToken
    refreshToken
    id
  }
}
```

2. Fetch all story translations using 
```
query StoryTranslations {
  storyTranslations(first: 13) {
    edges {
      cursor
      node {
        storyTranslationId
        title
        language
        stage
      }
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}
```

the `hasNextPage` value should be false, indicating that there are no more story translations left.

3. Run the same query as above except using `first: 2`, verify that two results are returned. Copy the `endCursor` string.
4. Run the query using `first: 2, after: cursor` where cursor is the copied value, verify that the next two story translations are returned.
5. Play around with the combination of filters and the pagination arguments, verify that the appropriate results are returned.

## What should reviewers focus on?
- Do the filters work properly with the pagination
- Does the pagination behave properly

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by jennchenn — COMMENTED** [review]
testing steps seemed to work for me!

what other arguments are supported? i tried using `offset` based on https://graphql.org/learn/pagination/#pagination-and-edges but couldn't get that to work so i just tested out calls with `first` and `after`

File: backend/python/app/graphql/types/story_type.py [hunk 1]
```diff
@@ -79,6 +79,32 @@ class StoryTranslationResponseDTO(graphene.ObjectType):
     reviewer_name = graphene.String()
 
 
+class StoryTranslationNode(graphene.ObjectType):
+    story_translation_id = graphene.Int(required=True)
```
[the remark this answers is not in the document]
Comment (gaoxk) [reply backend/python/app/graphql/types/story_type.py:83]: Unfortunately the Node type adds an id field that cant be overwritten - but I'll let philips explain

[the remark this answers is not in the document]
Comment (jennchenn) [reply backend/python/app/graphql/types/story_type.py:83]: yeah i noticed that too :(
just commenting so we remember to switch the .id !

**Review by gaoxk — APPROVED** [review]
noiceeeeeeeeeeeee 😎

File: backend/python/app/graphql/schema.py [hunk 2]
```diff
@@ -150,7 +154,7 @@ def resolve_story_translations_by_user(
         )
 
     def resolve_story_translations(
-        root, info, language=None, level=None, stage=None, story_title=None
+        root, info, language=None, level=None, stage=None, story_title=None, **kwargs
```
[the remark this answers is not in the document]
Comment (Puepis) [reply backend/python/app/graphql/schema.py:157]: hm I'm not sure if it's possible to set a default, i tried doing `first=10` in the arguments but it didn't work

like you said I think we can just add it on the frontend side

**Comment by Puepis** [comment]
> 
> 
> testing steps seemed to work for me!
> 
> what other arguments are supported? i tried using `offset` based on https://graphql.org/learn/pagination/#pagination-and-edges but couldn't get that to work so i just tested out calls with `first` and `after`

i think first, after, before, and last are the only supported ones - for our use case we'll prob only be using first and after

File: backend/python/app/graphql/types/story_type.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Puepis) [reply backend/python/app/graphql/types/story_type.py:83]: yup i'll make sure to change it along with the other frontend changes!

**Pull request merged**